File: FSST/OS/02-THREADS/02-ueben/ue-java/AB-02-java-threads-basics/OS00_PunktStrich.java
/**
 * Punkte und Striche werden abwechselnd gezeigt.
 * Der Scheduler ist für das Wechseln zuständig.
 */ 

/*
javac OS00_PunktStrich.java
java OS00_PunktStrich
*/

class PunktStrich
//
// ENTER CODE HERE
//


public class OS00_PunktStrich {
	public static void main(String[] args) {
	
		PunktStrich punkt= new PunktStrich(".");
		PunktStrich strich= new PunktStrich("-");
		
		punkt.start();   //!!!!!!!!!!!!!!!
		strich.start();  //!!!!!!!!!!!!!!!
		
		// läuft bis der User Ctrl-C eingibt
	}
}

File: FSST/courses-master/OS/02-THREADS/02-ueben/ue-cpp-threads-qt/02-cpp-uebungsaufgaben/OS00-PunktStrich.cpp
/**
 * Punkte und Striche werden abwechselnd gezeigt.
 * Der Scheduler ist für das Wechseln zuständig.
 */

#include <QThread>
#include <iostream>
using namespace std;

//                      !!!!!!!!!!!!!!!
class PunktStrichThread : public  QThread {
private:
	string s;
public:
	PunktStrichThread(string s){
		this->s= s;
	}
	// ***** ENTER CODE HERE ***** //
};

int main(int argc, char *argv[]){
	PunktStrichThread punkt(".");
	PunktStrichThread strich("-");

	punkt.start();   
	strich.start();  

	punkt.wait(5000); // wait 5 sec
	punkt.terminate();
	strich.terminate();

	return 0;
}
